pennylane/estimator/ops/op_math/symbolic.py

```diff
@@ -0,0 +1,473 @@
+        kwargs = {key: value for key, value in kwargs.items() if key not in base_params}
+
+        try:
+            return base_class.adjoint_resource_decomp(base_params)
+        except ResourcesUndefinedError:
+            gate_lst = []
+            decomp = base_class.resource_decomp(**base_params, **kwargs)
+
+            for gate in decomp[::-1]:  # reverse the order
+                gate_lst.append(_apply_adj(gate))
+            return gate_lst
+
+    @classmethod
+    def adjoint_resource_decomp(cls, target_resource_params: dict) -> list[GateCount]:
+        r"""Returns a list representing the resources for the adjoint of the operator.
+
+        Args:
+            target_resource_params (dict): A dictionary containing the resource parameters of the
+                target operator.
+
+        Resources:
+            The adjoint of an adjointed operation is just the original operation. The resources
+            are given as one instance of the base operation.
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects, where each object
+            represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        base_cmpr_op = target_resource_params.get("base_cmpr_op")
+        return [GateCount(base_cmpr_op)]
+
+    @staticmethod
+    # pylint: disable=arguments-renamed
+    def tracking_name(base_cmpr_op: CompressedResourceOp) -> str:
+        r"""Returns the tracking name built with the operator's parameters."""
+        base_name = base_cmpr_op.name
+        return f"Adjoint({base_name})"
+
+
+class Controlled(ResourceOperator):
+    r"""Resource class for the symbolic Controlled operation.
+
+    A symbolic class used to represent the application of some base operation controlled on the
+    state of some control qubits.
+
+    Args:
+        base_op (:class:`~.pennylane.estimator.resource_operator.ResourceOperator`): The base operator to be
+            controlled.
+        num_ctrl_wires (int): the number of qubits the operation is controlled on
+        num_zero_ctrl (int): the number of control qubits, that are controlled when in the
+            :math:`|0\rangle` state
+
+    Resources:
+        The resources are determined as follows. If the base operator implements the
+        :code:`.controlled_resource_decomp()` method, then the resources are obtained directly from
+        this.
+
+        Otherwise, the controlled resources are given in two steps. Firstly, any control qubits
+        which should be triggered when in the :math:`|0\rangle` state, are flipped. This corresponds
+        to an additional cost of two ``X`` gates per :code:`num_zero_ctrl`.
+        Secondly, the base operation resources are extracted and we add to the cost the controlled
+        variant of each operation in the resources.
```

Comment:
```suggestion
        The resources are determined as follows. If the base operator implements the
        :code:`.controlled_resource_decomp()` method, then the resources are obtained directly from
        this object. Otherwise, the controlled resources are given in two steps. Firstly, any control qubits
        which should be triggered when in the :math:`|0\rangle` state, are flipped. This corresponds
        to an additional cost of two ``X`` gates per :code:`num_zero_ctrl`.
        Secondly, the base operation resources are extracted and we add to the cost the controlled
        variant of each operation in the resources.
```